packages/core/content-manager/server/services/entity-manager.js

```diff
@@ -300,4 +300,23 @@ module.exports = ({ strapi }) => ({
 
     return sumDraftCounts(entity, uid);
   },
+
+  async countManyEntriesDraftRelations(ids, uid) {
+    const { populate, hasRelations } = getDeepPopulateDraftCount(uid);
+
+    if (!hasRelations) {
+      return 0;
+    }
+
+    const entities = await strapi.entityService.findMany(uid, {
+      populate,
+      where: { id: { $in: ids } },
+    });
+
+    const totalNumberDraftRelations = await reduceAsync(entities)(async (count, entity) => {
+      return sumDraftCounts(entity, uid) + count;
+    }, 0);
```

Comment:
with array reduce I always receive a Promise
I tried different combinatiions and I always had to add await to the accumulator

Reply:
I think it could work if you dont define the method as async

```js
const totalNumberDraftRelations = entities.reduce((count, entity) => sumDraftCounts(entity, uid) + count, 0);
```
you receive a promise like this?

Reply:
nice 🚀  probably you had the async "descriptor" in the reduce function
entities.reduce( async (count, entity) => sumDraftCounts(entity, uid) + count, 0);